internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1936,7 +1946,21 @@ func clusterRequirementsMet(clusterInstall *hiveext.AgentClusterInstall, status
 	})
 }
 
-func clusterCompleted(clusterInstall *hiveext.AgentClusterInstall, status, statusInfo string, opers []*models.MonitoredOperator) {
+func IsHoldInstallationSet(ci *hiveext.AgentClusterInstall, cd *hivev1.ClusterDeployment) bool {
+	if ci.Spec.HoldInstallation {
+		return true
+	}
+	if cd.Annotations != nil {
+		if paused, err := strconv.ParseBool(cd.Annotations[ReconcilePauseAnnotation]); err == nil && paused {
+			return true
```

Comment:
Maybe worth adding some log here? I.e. as an indication that installation is paused even though HoldInstallation is false.

Reply:
I added to the condition text https://github.com/openshift/assisted-service/pull/5549/files#diff-1dcba45c4f6078997bc41f1f68481cc5c952bbb45a34a59eeb361134382714e3R62

Reply:
I actually just meant logs for us to debug... So we could easily detect the reason for a paused installation when HoldInstallation is false. Not too crucial of course, be maybe might be helpful.

Reply:
a lot of things usually happen in parallel, looking at the logs  wasn't very helpful to me when debugging something like that, but sure no problem to add